ui/pages/confirmation/templates/add-ethereum-chain.js

```diff
@@ -331,7 +333,32 @@ function getValues(pendingApproval, t, actions, history) {
     ],
     cancelText: t('cancel'),
     submitText: t('approveButtonText'),
+    loadingText: t('addingCustomNetwork'),
     onSubmit: async () => {
+      let endpointChainId;
+      try {
+        endpointChainId = await jsonRpcRequest(customRpcUrl, 'eth_chainId');
+      } catch (_err) {
+        // eslint-disable-next-line no-console
+        console.error(
+          `Request for method 'eth_chainId on ${customRpcUrl} failed`,
+        );
+        return actions.rejectPendingApproval(
+          pendingApproval.id,
+          ethErrors.rpc.resourceUnavailable().serialize(),
+        );
+      }
+
+      if (pendingApproval.requestData.chainId !== endpointChainId) {
+        console.error(
+          `Chain ID returned by RPC URL ${customRpcUrl} does not match ${endpointChainId}`,
+        );
+        return actions.rejectPendingApproval(
```

Comment:
This is a clever solution to the double-window-popup problem :+1: Great work

The one major flaw left is that this experience has been made more confusing for the user because of this change, specifically in the case where this validation fails.

Previously if this validation step failed, the dapp would get an error and the user would see nothing. In this circumstance, maybe the dapp handled it well and maybe they didn't, but at least MetaMask isn't doing anything to mislead/confuse users.

But now if validation fails, it will appear to a user that the operation succeeded. Only the dapp and the web console are told of the error. The user may not find out.

It would be better to show the user the error somehow. But I don't know of an easy way to do this :thinking:

Reply:
Maybe this UX concern is not a blocker, and we can solve it in a later PR. We should at least track this problem as an issue if it's not solved here though.

Reply:
As a temporary workaround until the other work to support this is complete, can we infer if the operation failed by following up with a second check to see what chain ID we are using?

Similar to whats done in:

https://github.com/MetaMask/metamask-extension/blob/cdc1bce6884295e97c63ee9758646fc4287aa781/ui/components/app/network-display/network-display.js#L31-L34

Reply:
`state.metamask.provider` gives us access to the following to check the chain ID: 


<img width="162" alt="Screenshot 2023-02-15 at 4 34 24 PM" src="https://user-images.githubusercontent.com/15018469/219173355-db76a7eb-6869-45fe-ab62-3292c2919499.png">

Reply:
Update: Michał and I discussed offline and he has found another solution to this problem